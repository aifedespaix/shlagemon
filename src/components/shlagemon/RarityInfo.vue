<script setup lang="ts">
import type { HTMLAttributes } from 'vue'

interface Props {
  /** Rarity value between 1 and 100. */
  rarity: number
}

defineOptions({ inheritAttrs: false })
const props = defineProps<Props>()
const attrs = useAttrs() as HTMLAttributes
const { t } = useI18n()
</script>

<template>
  <UiTooltip :text="t('components.shlagemon.RarityInfo.tooltip')">
    <ShlagemonRarity :rarity="props.rarity" v-bind="attrs" />
  </UiTooltip>
</template>
