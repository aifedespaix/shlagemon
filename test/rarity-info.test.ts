import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import { createI18n } from 'vue-i18n'
import RarityInfo from '../src/components/shlagemon/RarityInfo.vue'

describe('component RarityInfo.vue', () => {
  it('renders rarity value with tooltip', () => {
    const i18n = createI18n({
      legacy: false,
      locale: 'en',
      messages: { en: { components: { shlagemon: { RarityInfo: { tooltip: 'Rarity' } } } } },
    })

    const wrapper = mount(RarityInfo, {
      props: { rarity: 42 },
      global: { plugins: [i18n] },
    })

    expect(wrapper.text()).toContain('42')
    expect(wrapper.html()).toMatchSnapshot()
  })
})
