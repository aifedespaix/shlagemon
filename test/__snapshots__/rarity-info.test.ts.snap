// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`component RarityInfo.vue > renders rarity value with tooltip 1`] = `"<div data-v-c07c8769="" class="text-grey-900 dark:text-grey-300 bg-blue-300 dark:bg-blue-800 aspect-square h-10 w-10 flex items-center justify-center rounded-full px-2 py-1">42</div>"`;
